libc/string_lenchr.c

```diff
@@ -150,9 +154,291 @@ TEST(string_len, big)
+
+	TEST_ASSERT_EQUAL_STRING(&whites[sz / 2], strchr(whites, whites[sz / 2]));
+	TEST_ASSERT_EQUAL_STRING(&whites[sz / 2], strrchr(whites, whites[sz / 2]));
+	TEST_ASSERT_EQUAL_STRING(&whites[sz / 2], memchr(whites, whites[sz / 2], sz));
+
+	TEST_ASSERT_EQUAL_STRING(&whites[0], strchr(whites, whites[0]));
+	TEST_ASSERT_EQUAL_STRING(&whites[0], strrchr(whites, whites[0]));
+	TEST_ASSERT_EQUAL_STRING(&whites[0], memchr(whites, whites[0], sz));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr(whites, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr(whites, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(whites, 'x', sz));
+}
+
+
+TEST(string_chr, empty)
+{
+	int sz;
+
+	/* Checking if we can get an empty string */
+	sz = 1;
+	TEST_ASSERT_EQUAL_STRING("", strchr("", '\0'));
+	TEST_ASSERT_EQUAL_STRING("", strrchr("", '\0'));
+	TEST_ASSERT_EQUAL_STRING("", memchr("", '\0', sz));
+
+	TEST_ASSERT_EQUAL_STRING("", strchr("", 0));
+	TEST_ASSERT_EQUAL_STRING("", strrchr("", 0));
+	TEST_ASSERT_EQUAL_STRING("", memchr("", 0, sz));
+
+	/* Looking for white spaces in empty string */
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr("", ' '));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr("", ' '));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr("", ' ', sz));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr("", '\n'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr("", '\n'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr("", '\n', sz));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr("", '\t'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr("", '\t'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr("", '\t', sz));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr("", '\r'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr("", '\r'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr("", '\r', sz));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr("", '\v'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr("", '\v'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr("", '\v', sz));
+}
+
+
+TEST(string_chr, memchr_size)
+{
+	char lorem[] = "Lorem";
+
+	/* Testing capability of setting size in memchr */
+	TEST_ASSERT_EQUAL_STRING("rem", memchr(lorem, 'r', 6));
+	TEST_ASSERT_EQUAL_STRING("rem", memchr(lorem, 'r', 5));
+	TEST_ASSERT_EQUAL_STRING("rem", memchr(lorem, 'r', 4));
+	TEST_ASSERT_EQUAL_STRING("rem", memchr(lorem, 'r', 3));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(lorem, 'r', 2));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(lorem, 'r', 1));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(lorem, 'r', 0));
+}
+
+
+/*
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
 TEST_GROUP_RUNNER(string_len)
 {
 	RUN_TEST_CASE(string_len, ascii);
 	RUN_TEST_CASE(string_len, not_ascii);
 	RUN_TEST_CASE(string_len, big);
 }
+
+
+TEST_GROUP_RUNNER(string_chr)
+{
+	RUN_TEST_CASE(string_chr, basic);
+	RUN_TEST_CASE(string_chr, big);
+	RUN_TEST_CASE(string_chr, special);
+	RUN_TEST_CASE(string_chr, ascii);
+	RUN_TEST_CASE(string_chr, not_ascii);
+	RUN_TEST_CASE(string_chr, torn);
+	RUN_TEST_CASE(string_chr, whitespaces);
```

Comment:
Please move the whitespaces case between `special` and `ascii`. Now we have tested whitespaces in the `ascii` case, so it's not so valuable. We want to test some basic stuff, special characters and whites in the separate cases before the whole ascii set to eventually catch smaller issues (for example if handling white spaces didn't work we want to see error of `whitespaces` test case, not `ascii`, which would also fail)